### DISCUSSION TITLE
Prompt flow docker to production

### DISCUSSION THREAD

**Discussion by AlexCastroAlex** [microsoft/promptflow#2815] [Q&A]
Hello all , 

I'm pretty new to prompt flow and maybe i did not find the information but as prompt flow makes API endpoints , is there a way to generation  OPEN API  manifests and swagger json/yaml  file ? 
The goal of this is to include this as plugin in many solutions.

Last question , i did not find where i can edit the name of the endpoint deployed ? Can someone give me a hint please ? 

Thanks in advance.

Regards

**Comment by brynn-code** [comment]
Hi, for the swagger file question: you could get the swagger via GET `<endpoint>/swagger.json`.
For the endpoint deploy question: Where did you deploy your flow to? I'm not sure what's the 'name of the endpoint' refer to.

**Reply by brynn-code** [reply]
I've added an issue to track the api doc progress and assigned to myself: #2837

**Reply by AlexCastroAlex** [reply]
First of all thanks for reply !
I deployed it on app service with Docker image and all i have in swagger.json is 2 endpoints : 
- /score
- /feedback
I wanted to know how to edit those two paths.

About the manifest , the file for plugin management the file should look like this : 
```json
{
    "schema_version": "v1",
    "name_for_human": "TODO List",
    "name_for_model": "todo",
    "description_for_human": "Manage your TODO list. You can add, remove and view your TODOs.",
    "description_for_model": "Help the user with managing a TODO list. You can add, remove and view your TODOs.",
    "auth": {
        "type": "none"
    },
    "api": {
        "type": "openapi",
        "url": "https://example.com/openapi.yaml"
    },
    "logo_url": "https://example.com/logo.png",
    "contact_email": "support@example.com",
    "legal_info_url": "http://www.example.com/legal"
}
```
this file has to be hosted also in a specific path /.well-known/your-manifest.json

This would be awesome to be able to generate this file and expose it because prompt flow could act as a plugin creator :)

As a workaround , maybe you could just add the opportunity to expose the file manually created in this specific path ? 

Thanks in advance.

Regards

**Reply by brynn-code** [reply]
Seems the example is from https://github.com/openai/plugins-quickstart/tree/main/.well-known, sorry that we don't support custom path for now, but your workaround suggestion is an interesting approach, we will consider about it.

**Reply by alexandrecastro-bib** [reply]
Thanks a lot for reply and consideration! 
The issue is that we let prompt flow sdk manage all the api stuff wirhout any custom settings available.
I’ll try to find to fond where api is generated and add a workaround for this.
if you have Time to indicate where i need to search, i’ll gain Time 😉

regards

**Reply by brynn-code** [reply]
😆 The api now located at here: https://github.com/microsoft/promptflow/blob/main/src/promptflow-core/promptflow/core/_serving/v1/app.py

**Reply by AlexCastroAlex** [reply]
Thanks !!  i'll go work on it :)

**Reply by AlexCastroAlex** [reply]
Hello @brynn-code , 

I made the changes on my fork but don't know how to test the generated API has the new endpoint (/.well-known/ai-plugin.json)

Is there a documentation to generate and use a new prompt flow version  ? 

Regards

**Reply by brynn-code** [reply]
Hi @AlexCastroAlex , the flask app can be start by command 'pf flow serve', you could reach this doc for more details: https://microsoft.github.io/promptflow/how-to-guides/deploy-a-flow/deploy-using-dev-server.html

Also, please note that you'll need to install dev version promptflow, you could run [dev setup](https://github.com/microsoft/promptflow/blob/main/scripts/dev-setup/main.py) here to setup env (as pip install -e directly may cause some problem).

**Reply by AlexCastroAlex** [reply]
Thanks for response @brynn-code  ! 

Feature has been actively tested and pull request has been made : https://github.com/microsoft/promptflow/pull/2895 

Regards
